mail_operating_unit/models/mail_mail_ext.py

```diff
@@ -24,18 +24,33 @@ def create(self, vals_list):
                 record = self.env[model].browse(res_id)
                 if getattr(record, 'operating_unit_id', False):
                     operating_unit = record.operating_unit_id
-            if not operating_unit and self.env.context.get('uid', False):
-                user = self.env['res.users'].browse(self.env.context.get('uid'))
+            if not operating_unit:
+                if self.env.context.get('uid', False):
+                    user = self.env['res.users'].browse(
+                        self.env.context.get('uid')
+                    )
+                else:
+                    user = self.env.user
                 operating_unit = user.operating_unit_for_mails_id
 
-            if operating_unit and operating_unit.catchall_alias and operating_unit.catchall_domain:
-                sender_email_address = (operating_unit.catchall_alias + '@' +
-                                        operating_unit.catchall_domain)
+            if operating_unit and operating_unit.catchall_alias and \
```

Comment:
Use parenthesis instead of the back slash in order to split IFs in multiple lines.